### ISSUE TITLE
Improper override not reported

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#6010]
This comes from a [mypy issue](https://github.com/python/mypy/issues/16165).

Here's a minimal repro:

```python
from __future__ import annotations
from typing import Self, TypeVar, Generic

T = TypeVar("T")

class Base:
    def __and__(self: Self, other: Self) -> Self:
        ...

class Derived(Generic[T], Base):
    def __and__(self: Derived[bool], other: Derived[bool]) -> Derived[bool]:
        return Derived[bool]()
```

**Comment by erictraut** [comment]
This is included in pyright 1.1.332, which I just published. It will also be included in a future release of pylance.
